### PR TITLE
bugfix: set reduction attribute as none on loss fn if nothing is specified

### PR DESCRIPTION
**Pull request by bharadwaj6**
Possible fix for #1136 

Previously it was passed as a kwarg to the loss function, but mixup loss function doesn't have setters for the same.
Pytorch core loss functions seem to import from `nn._Loss` which has the setters for `reduction` parameter, so inheriting `MixupLoss` from that instead of `nn.Module` seems like another way to fix this bug. 

Setting the attribute explicitly when it doesn't exist seemed like a better approach, which is why the changes.

### CODE REVIEW

**Comment by sgugger** [comment]
Not sure this is the right way to do it. I think the reduction attribute should be passed to the loss function inside MixupLoss instead. Closing this, the issue is there and I'll try to solve it today or tomorrow.

**Comment by bharadwaj6** [comment]
@sgugger Have created another PR with suggested changes at #1163. Hope I got the details right this time.
